app/services/deployment/processes/refresh_project_instance_configuration.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Deployment
+  module Processes
+    class RefreshProjectInstanceConfiguration
+      def initialize(project_instance)
+        @project_instance = project_instance
+      end
+
+      def call
+        @project_instance.configurations.each do |configuration|
+          repository = Repository.find(configuration.repository_id)
+          configuration.env_variables = repository.build_env_variables
```

Comment:
we have `Deployment::ConfigurationBuilders::EnvVariablesBuilder` class - take a look. Repository doesn't stores all variables